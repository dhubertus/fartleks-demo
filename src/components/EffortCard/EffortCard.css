.effort-card {
  height: 75px;
  width: 16%;
  border: 1px solid #3d3e42;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background: #fff;
  border-radius: 5px;
  color: #00aced;
  font-weight: 100;
  /* box-shadow: 1px 1px 1px 1px black; */
}

.effort-card:hover {
  /* border: 1px solid #00aced; */
  cursor: pointer;
  color: #ffa100;
}
