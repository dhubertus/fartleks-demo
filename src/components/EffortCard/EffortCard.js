import React, { Component } from 'react';

import HandleDataApi from '../../helpers/HandleDataApi';
import DataSelectionContainer from '../../containers/DataSelectionContainer';

export class EffortCard extends Component {
  render() {
    const { title, bestAverage, bestValues } = this.props;
    const api = new HandleDataApi();
    const readyToMapData = api.getAllGPSLocations(bestValues);

    return (
      <div className="effort-card" onClick={() => this.props.handleSelectedData(readyToMapData)}>
        <div>{`${title} Minute`.toUpperCase()}</div>
        <div>{bestAverage.toFixed(2)}</div>
      </div>
    );
  }
}

export default DataSelectionContainer(EffortCard);
