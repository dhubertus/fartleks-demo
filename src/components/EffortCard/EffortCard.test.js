import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import { EffortCard } from './EffortCard';

describe('Effort Card Tests', () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const mockFn = jest.fn();
  const initialState = {
    selectedData: [],
    bestEfforts: {},
  };
  const store = mockStore(initialState);
  const wrapper = shallow(<EffortCard
                            store={store}
                            key={0}
                            title={'ten'}
                            bestAverage={100}
                            bestValues={[]}
                            handleSelectedData={mockFn}
                          />);

  it('should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a single effort card', () => {
    wrapper.find('.effort-card').simulate(
      'click',
      { preventDefault() {} },
    );
    expect(wrapper.find('.effort-card').length).toBe(1);
  });

  it('should trigger this.props.handleSelectedData when clicked', () => {
    expect(mockFn.mock.calls.length).toBe(1);
  });
});
