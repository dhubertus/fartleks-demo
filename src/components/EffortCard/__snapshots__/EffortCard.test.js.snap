// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Effort Card Tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EffortCard
    bestAverage={100}
    bestValues={Array []}
    handleSelectedData={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    title="ten"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          TEN MINUTE
</div>,
        <div>
          100.00
</div>,
      ],
      "className": "effort-card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TEN MINUTE",
        },
        "ref": null,
        "rendered": "TEN MINUTE",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "100.00",
        },
        "ref": null,
        "rendered": "100.00",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            TEN MINUTE
</div>,
          <div>
            100.00
</div>,
        ],
        "className": "effort-card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TEN MINUTE",
          },
          "ref": null,
          "rendered": "TEN MINUTE",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "100.00",
          },
          "ref": null,
          "rendered": "100.00",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
